packages/api/internal/handlers/template_build_logs.go

```diff
@@ -46,27 +48,21 @@ func (a *APIStore) GetTemplatesTemplateIDBuildsBuildIDStatus(c *gin.Context, tem
 	if err != nil {
 		errMsg := fmt.Errorf("error when parsing build id: %w", err)
 		telemetry.ReportError(ctx, errMsg)
-
 		a.sendAPIStoreError(c, http.StatusBadRequest, "Invalid build id")
-
 		return
 	}
 
-	dockerBuild, err := a.buildCache.Get(templateID, buildUUID)
+	templateDB, _, err := a.db.GetEnv(ctx, templateID)
```

Comment:
sorry for spotting it now, but i think it could make sense to keep the cache (as an actual cache), probably with lower TTL. There's 2 db request every call, it's probably unnecessary. Also once the build existed, there's no reason why you should be able to query logs, so it's fairly safe (I would still keep some mechanism to recheck DB as we do e.g. for auth)